### PR TITLE
Add new feature for custom workspace network CIDR

### PR DESCRIPTION
**Pull request by aledbf**
## Description

The default workspace network cell uses a private range that can conflict with some pre-existing network in the VPC that creates a network routing error (`no route to host`) when we try to reach any IP address of the such subnet.
For this scenario, we introduce a new ws-daemon parameter that allows us to set a different subnet. If the parameter is not present, we use the default value we use.

fixes WKS-50

## How to test
- Use the default configuration
  - Verify workspaces run
  - Check the IP addresses in the workspace. The value should be `10.0.5.2/30`
- Change the installer configuration to set a different subnet
  - Verify workspaces run
  - Check the IP addresses in the workspace. The value should match the subnet, and the workspace should have the second IP address in the range.

## Release Notes
```release-note
NONE
```

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=workspace
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by roboquat** [comment]
@aledbf: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/ws-daemon/nsinsider/main.go [hunk 1]
```diff
@@ -268,22 +269,25 @@ func main() {
...
+					vethIp, cethIp, mask, err := processWorkspaceCIDR(networkCIDR)
+					if err != nil {
+						return xerrors.Errorf("parsing workspace CIDR (%v):%v", networkCIDR, err)
 					}
-					masqueradeAddr := net.IPNet{
-						IP:   vethIp.IP.Mask(mask),
-						Mask: mask,
+
+					vethIpNet := net.IPNet{
+						IP:   net.ParseIP(vethIp.String()),
```
Comment (Furisto) [inline components/ws-daemon/nsinsider/main.go:287]: vethIp is already a net.IP

```suggestion
						IP:  IP:   vethIp,
```

File: components/ws-daemon/nsinsider/main.go [hunk 2]
```diff
@@ -429,23 +413,31 @@ func main() {
...
+					vethIp, cethIp, mask, err := processWorkspaceCIDR(networkCIDR)
+					if err != nil {
+						return xerrors.Errorf("parsing workspace CIDR (%v):%v", networkCIDR, err)
 					}
-					vethIp := net.IPNet{
-						IP:   net.IPv4(10, 0, 5, 1),
-						Mask: mask,
+
+					cethIpNet := net.IPNet{
+						IP:   net.ParseIP(cethIp.String()),
```
Comment (Furisto) [inline components/ws-daemon/nsinsider/main.go:432]:
```suggestion
						IP:  cethIP,
```

**Review by Furisto — APPROVED** [review]

**Comment by Furisto** [comment]
LGTM but holding in case you want to take care of the comments.

File: components/ws-daemon/nsinsider/main.go (hunk 1 above)
Comment (aledbf) [reply components/ws-daemon/nsinsider/main.go]: done

File: components/ws-daemon/nsinsider/main.go (hunk 2 above)
Comment (aledbf) [reply components/ws-daemon/nsinsider/main.go]: done

**Comment by aledbf** [comment]
Running additional tests before merging

**Comment by aledbf** [comment]
/werft run with-clean-slate-deployment=true with-preview=true recreate-vm=true

**Pull request merged**